src/libs/Navigation/Navigation.js

```diff
@@ -36,15 +44,16 @@ function goBack() {
  * @param {String} route
  */
 function navigate(route = ROUTES.HOME) {
-    // If we're navigating to the signIn page, replace the existing route in the stack with the SignIn route so that we
-    // don't mistakenly route back to any older routes after the user signs in
-    if (route === ROUTES.SIGNIN) {
-        navigationRef.current.dispatch(StackActions.replace(SCREENS.SIGN_IN));
-        return;
-    }
-
     if (route === ROUTES.HOME) {
-        openDrawer();
+        if (isLoggedIn) {
+            openDrawer();
+            return;
+        }
+
+        // If we're navigating to the signIn page, replace the existing route in the stack with the SignIn route
+        // so that we don't mistakenly route back to any older routes after the user signs in
+        navigationRef.current.dispatch(StackActions.replace(SCREENS.HOME));
+        navigationRef.current.dispatch(StackActions.popToTop());
```

Comment:
I'm not sure this is correct.

I logged out the `state` object when hitting the `ValidateLogin` screen and saw that it had this shape:

```
"routes": [
    {"key": "Home-WcWMk5JnZsVuWniETSMSs", "name": "Home", "params": undefined}, 
    {"key": "ValidateLogin-sIUmsR8Hy6WyQbAnYSjsu", "name": "ValidateLogin", "params": ... }
]
```

The `SignIn` screen is _already_ underneath `ValidateLogin` (since we named it `Home`). So, I think we just need to dismiss the `ValidateLogin` screen?